main.cpp

```diff
@@ -258,271 +282,316 @@ struct MotorCycle
     std::string memoryType = "RAM";
-    // RAM size
     unsigned int ramSize = 512;
-    // number of input ports
     unsigned int numberInputPorts = 4;
 
-    // fetch instructions from memory
-    std::string fetchInstructionsFromMemory(std::string memoryType = "RAM"); // get next instruction
-    // decode instructions into command
-    unsigned int decodeInstruction(std::string instruction = "Iddle"); // map instruction to unique number
-    // execute command
+    std::string fetchInstructionsFromMemory(std::string memoryType = "RAM");
+    unsigned int decodeInstruction(std::string instruction = "Iddle");
     void executeCommand(unsigned int numericCommand = 0);
     
 };
 
-/*
-Thing 10) Hearing aid
-5 properties:
-    1) Speakers
-    2) Microphone
-    3) Antenna
-    4) MCU
-    5) Battery
-3 things it can do:
-    1) amplify audio signals
-    2) listen to music
-    3) connect to smartphone
- */
+std::string MicroControllerUnit::fetchInstructionsFromMemory(std::string memoryTypeA)
+{
+    std::string instruction;
+    memoryType = memoryTypeA;
+    
+    if(memoryType == "RAM")
+        instruction = "ABAB";
+    else if(memoryType == "ROM")
+        instruction = "CDCD";
+    else
+        instruction = "Iddle";
+
+    return instruction;
+}
+
+unsigned int MicroControllerUnit::decodeInstruction(std::string instruction)
+{
+    unsigned int decodedInstruction;
+
+    if(instruction == "ABAB")
+        decodedInstruction = 3;
+    else if(instruction == "CDCD")
+        decodedInstruction = 2;
+    else if(instruction == "Iddle")
+        decodedInstruction = 1;
+    else
+        decodedInstruction = 0;
+
+    return decodedInstruction;
+}
 
 struct HearingAid
 {
-    // Speakers
     Speaker speaker;
-    // Microphone
     Microphone mic;
-    // Antenna
     Antenna antenna;
-    // MCU
-    MicrocontrollerUnit microController;
-    // Battery
+    MicroControllerUnit microController;
     Battery battery;
 
-    // amplify audio signals
-    void amplifyAudioSignals(Microphone microphoneA, float gain); // 4) some of input parameters is a UDT
-    // listen to music
-    void listenToMusic(Speaker speakerA, float gain);
-    // connect to smartphone. 4) some of input parameters is a UDT
-    bool connectToSmartphone(std::string handshakingID = "AAAA"); // confirm whether or not the conenction to a smartphone was succesful
+    void amplifyAudioSignals(Microphone microphoneA, float gain);
+    void listenToMusic();
+    bool connectToSmartphone(std::string handShakingID = "ABAB");
     
 };
 
+void HearingAid::amplifyAudioSignals(Microphone microphoneA, float gain)
+{
+    microphoneA.setInputGain(gain);
+}
+
+void HearingAid::listenToMusic()
+{
+    speaker.setPlaybackLevel(8);
+    speaker.playSound();
+}
+
+bool HearingAid::connectToSmartphone(std::string handShakingID)
+{
+    unsigned int numericInstruction;
+    
+    numericInstruction = microController.decodeInstruction(handShakingID);
```

Comment:
Don't declare the variable separately from where you initialize it. Do it all in one line.
```
float name = expression;
```
- [ ] revise